### ISSUE TITLE
panic when field of type map[string]any is in the struct

### ISSUE THREAD

**Issue by lev112** [go-faker/faker#30]
given this struct
```
type MyStruct struct {
	A string
	B map[string]any
}
```

the following code will fail 
```
var base MyStruct
err := faker.FakeData(&base) 
```
with an error:
> interface{} not allowed

(this is how I would expect it to behave)

if I try to ignore the fields that include an interface:
```
var base MyStruct
err := faker.FakeData(&base,
	options.WithIgnoreInterface(true),
)
```
I would expect the faking to work, and leave the `B` field empty,
but the code will panic with
> panic: runtime error: invalid memory address or nil pointer dereference

---
adding 
```
options.WithFieldsToIgnore("B")
```
will solve that problem, but the panic is very confusing, 
and there is no hint of what is the problem

**Comment by gegaryfa** [comment]
Hi @lev112 , which version are you using?
I have a similar issue after upgrading to 4.1.1 which includes the changes of this PR #25.
Up to 4.0.0 works fine without panic

**Comment by lev112** [comment]
@gegaryfa I'm using the latest (4.1.1)

**Comment by gegaryfa** [comment]
@bxcodec is this intended behaviour?

**Comment by bxcodec** [comment]
Hi @lev112  and @gegaryfa I fixed it in #36 
Will release a new version soon, please help to check again with your use case, thanks. 
*feel free to re-open this issue if it's still happening
